bottlecap/src/bin/bottlecap/main.rs

```diff
@@ -425,8 +425,9 @@ async fn extension_loop_active(
                                     ..
                                 } => {
                                     let mut p = invocation_processor.lock().await;
+                                    let mut enhanced_metric_data = None;
                                     if let Some(metrics) = metrics {
-                                        p.on_platform_runtime_done(
+                                        enhanced_metric_data = p.on_platform_runtime_done(
                                             &request_id,
                                             metrics.duration_ms,
                                             config.clone(),
```

Comment:
all these stuff like configs and tags provider could be set once at initialization of the invocation_processor, rather than cloned continuously

Reply:
True, I'll be doing some refactoring in my next PR, so I can look into that then